api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -193,6 +193,7 @@ func (s *SolutionManager) Reconcile(ctx context.Context, deployment model.Deploy
 		context.DeploymentSpec = deployment
 		context.Value = deployment
 		context.Component = ""
+		context.Scope = scope
```

Comment:
This is interesting. Could you please help me understand why do you set context.Scope here?

Reply:
That's the whole point of the PR. See the title. When we're resolving $config() expression in a solution or a catalog, we want to find a catalog in the same namespace. So we're saving the current scope in the evaluationContext (this knowledge will be useful in future in other expressions). This context is [passed by the parser](https://github.com/eedorenko/eclipse-symphony/blob/467f7a737928b15644356f8af2a30c2996492da7/api/pkg/apis/v1alpha1/utils/parser.go#L775) to the provider, which eventually brings it to the [Read](https://github.com/eedorenko/eclipse-symphony/blob/467f7a737928b15644356f8af2a30c2996492da7/api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go#L104) and [ReadObject](https://github.com/eedorenko/eclipse-symphony/blob/467f7a737928b15644356f8af2a30c2996492da7/api/pkg/apis/v1alpha1/providers/config/catalog/catalogprovider.go#L128C33-L128C43) methods. Those guys extract the scope from the context and pass it to the [utils.GetCatalog](https://github.com/eedorenko/eclipse-symphony/blob/467f7a737928b15644356f8af2a30c2996492da7/api/pkg/apis/v1alpha1/utils/symphony-api.go#L148) method. Because the signature of the utils.GetCatalog has been enriched with the scope parameter, the rest of the files in this PR have been affected.